### PR TITLE
Feature/#704 add copy to clipboard button in application view

### PR DESCRIPTION
**Pull request by tomlovesgithub**
- Add function for copying panel-pack content to clipboard
- Add button, with text "Copy to clipboard" which triggers function onClick
- Button text changes to "Copied" for 3 seconds after click, then returns to "Copy to clipboard"

-Did play around with positioning of button, needs either to be outside of tabs area (like pdf and doc buttons), so that the button remains visible when tabs are clicked or inside each tab.

### CODE REVIEW

File: src/views/Exercises/Applications/Application.vue [hunk 1]
```diff
@@ -97,6 +97,17 @@
             </div>
           </div>
         </div>        
+        <div class="text-right">
+          <!-- <a> -->
+          <button
```
Comment (YaaL) [inline src/views/Exercises/Applications/Application.vue:102]: Can we keep this button together with the other buttons?

File: src/views/Exercises/Applications/Application.vue [hunk 2]
```diff
@@ -1858,13 +1869,26 @@ export default {
...
 
       pdf.save(`${fileName}.pdf`);
     },
+    copyToClipboard() {
+      const content = document.querySelector('#panel-pack-div').innerText;
+      const el = document.createElement('textarea');
+      el.value = content;
+      document.body.appendChild(el);
+      el.select();
+      document.execCommand('copy');
```
Comment (YaaL) [inline src/views/Exercises/Applications/Application.vue:1882]: Almost perfect but the copied text contains page elements:

> Application
> JAC00216-lax0001
> **Download As PDFDownload As DocUnlock**
> Created on
> 20 April 2020, 14:54
> Submitted on
> 21 April 2020, 17:33
> **Copy to clipboard**
> **Full information**
> **Panel pack**
> **Issues**
> **Agency report**
> Personal details
> **Any changes made here will also update the candidate information.**
> Full Name	
> Claire Troughton
> **Change**
> ...

**Comment by tomlovesgithub** [comment]
Please excuse the atrocity above, i tried to squash a commit with `rebase -i` ...wont try that again

File: src/views/Exercises/Applications/Application.vue [hunk 3]
```diff
@@ -1864,6 +1871,29 @@ export default {
 
       pdf.save(`${fileName}.pdf`);
     },
+    copyToClipboard() {
+      let content = document.querySelector('#panel-pack-div').innerText.split('\n');
+      content.splice(2,1);
```
Comment (YaaL) [inline src/views/Exercises/Applications/Application.vue:1876]: This is very hacky, relying on line breaks always being in the same position!

File: src/views/Exercises/Applications/Application.vue [hunk 4]
```diff
@@ -1864,6 +1871,29 @@ export default {
 
       pdf.save(`${fileName}.pdf`);
     },
+    copyToClipboard() {
+      let content = document.querySelector('#panel-pack-div').innerText.split('\n');
```
Comment (YaaL) [inline src/views/Exercises/Applications/Application.vue:1875]: Try to copy the whole thing, remove unnecessary elements (some are currently marked by class `print-none`) and then grab the text.

Comment (warrensearle) [reply src/views/Exercises/Applications/Application.vue:1875]: Stripping out all the elements with `print-none` class sounds like a great approach 👍

File: src/views/Exercises/Applications/Application.vue [hunk 5]
```diff
@@ -1864,6 +1870,35 @@ export default {
...
+        if (htmlCollection.children.length){
+          Array.from(htmlCollection.children).forEach(element => {
+            result.push(this.removePrintNone(element));
+          });
+        } 
+        result.push(htmlCollection.innerText);
+      }
+      return result;
+    },
+    copyToClipboard() {
```
Comment (warrensearle) [inline src/views/Exercises/Applications/Application.vue:1885]: This looks lonely. Nice recursive function but can be removed, right?

File: src/views/Exercises/Applications/Application.vue [hunk 6]
```diff
@@ -73,10 +79,8 @@
           <div class="govuk-grid-column-one-half">
             <div class="panel govuk-!-margin-bottom-9 govuk-!-padding-4 background-light-grey">
               <span class="govuk-caption-m">Created on</span>
-              <h2 
-                class="govuk-heading-m govuk-!-margin-bottom-0"
-              >
-                {{ application.createdAt | formatDate | showAlternative("Unknown") }}
+              <h2 class="govuk-heading-m govuk-!-margin-bottom-0">
+                {{ new Date(application.createdAt) | formatDate('longdatetime') }}
```
Comment (YaaL) [inline src/views/Exercises/Applications/Application.vue:83]: Oh no, we need the other one here

File: src/views/Exercises/Applications/Application.vue [hunk 7]
```diff
@@ -88,7 +92,7 @@
                 v-if="isApplied"
                 class="govuk-heading-m govuk-!-margin-bottom-0"
               >
-                {{ application.appliedAt | formatDate | showAlternative("Unknown") }}
+                {{ application.appliedAt | formatDate('longdatetime') }}
```
Comment (YaaL) [inline src/views/Exercises/Applications/Application.vue:95]: Need the other one here too

File: src/views/Exercises/Applications/Application.vue [hunk 8]
```diff
@@ -33,20 +33,44 @@
...
-              class="govuk-button govuk-button--secondary"
-              @click="downloadAsDoc"
-            >
-              Download As Doc
-            </button>
-
+            <div class="moj-button-menu">
+              <button
+                ref="dropDownRef"
+                class="govuk-button govuk-button govuk-button--secondary moj-button-menu__toggle-button"
```
Comment (YaaL) [inline src/views/Exercises/Applications/Application.vue:39]: Add `moj-button-menu__toggle-button--secondary` here to make the arrow black (it's currently white)

Comment (tomlovesgithub) [reply src/views/Exercises/Applications/Application.vue:39]: ahhhhhhhh, missed the toggle menu bit!

**Pull request merged**